Jackson version incompatibility

## This is a...

<pre><code>
[ ] Feature request
[x] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Documentation issue or request
</code></pre>

## Description

Seems that the change in #6914 triggered a [nightly build failure](https://circleci.com/gh/syndesisio/syndesis/157090). We most likely are using two different versions of Jackson between Syndesis and integrations.

Perhaps it would be a good idea to remove dependency on syndesis/common-util from the integration runtime/connectors.

```
01:16:58.649 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT: Caused by: java.lang.NoSuchMethodError: com.fasterxml.jackson.databind.ObjectMapper.setDefaultPropertyInclusion(Lcom/fasterxml/jackson/annotation/JsonInclude$Value;)Lcom/fasterxml/jackson/databind/ObjectMapper;
01:16:58.649 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at io.syndesis.common.util.Json.<clinit> (Json.java:45)
01:16:58.650 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at io.syndesis.integration.runtime.IntegrationRouteBuilder.loadIntegration (IntegrationRouteBuilder.java:115)
01:16:58.650 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at io.syndesis.integration.runtime.IntegrationRouteBuilder.configure (IntegrationRouteBuilder.java:140)
01:16:58.651 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.apache.camel.builder.RouteBuilder.checkInitialized (RouteBuilder.java:462)
01:16:58.651 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.apache.camel.builder.RouteBuilder.configureRoutes (RouteBuilder.java:402)
01:16:58.651 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.apache.camel.builder.RouteBuilder.addRoutesToCamelContext (RouteBuilder.java:383)
01:16:58.654 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.apache.camel.impl.DefaultCamelContext$1.call (DefaultCamelContext.java:1027)
01:16:58.655 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.apache.camel.impl.DefaultCamelContext$1.call (DefaultCamelContext.java:1024)
01:16:58.655 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.apache.camel.impl.DefaultCamelContext.doWithDefinedClassLoader (DefaultCamelContext.java:3270)
01:16:58.656 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.apache.camel.impl.DefaultCamelContext.addRoutes (DefaultCamelContext.java:1024)
01:16:58.656 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at io.syndesis.integration.runtime.sb.IntegrationRuntimeAutoConfiguration$1.beforeApplicationStart (IntegrationRuntimeAutoConfiguration.java:93)
01:16:58.657 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.apache.camel.spring.boot.RoutesCollector.onApplicationEvent (RoutesCollector.java:152)
01:16:58.657 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.apache.camel.spring.boot.RoutesCollector.onApplicationEvent (RoutesCollector.java:57)
01:16:58.657 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener (SimpleApplicationEventMulticaster.java:172)
01:16:58.660 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener (SimpleApplicationEventMulticaster.java:165)
01:16:58.660 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent (SimpleApplicationEventMulticaster.java:139)
01:16:58.661 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.context.support.AbstractApplicationContext.publishEvent (AbstractApplicationContext.java:393)
01:16:58.661 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.context.support.AbstractApplicationContext.publishEvent (AbstractApplicationContext.java:399)
01:16:58.661 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.context.support.AbstractApplicationContext.publishEvent (AbstractApplicationContext.java:347)
01:16:58.663 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.context.support.AbstractApplicationContext.finishRefresh (AbstractApplicationContext.java:883)
01:16:58.663 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.finishRefresh (EmbeddedWebApplicationContext.java:144)
01:16:58.664 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.context.support.AbstractApplicationContext.refresh (AbstractApplicationContext.java:546)
01:16:58.664 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh (EmbeddedWebApplicationContext.java:122)
01:16:58.665 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.boot.actuate.autoconfigure.EndpointWebMvcAutoConfiguration.createChildManagementContext (EndpointWebMvcAutoConfiguration.java:193)
01:16:58.667 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.boot.actuate.autoconfigure.EndpointWebMvcAutoConfiguration.afterSingletonsInstantiated (EndpointWebMvcAutoConfiguration.java:156)
01:16:58.668 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons (DefaultListableBeanFactory.java:781)
01:16:58.668 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization (AbstractApplicationContext.java:867)
01:16:58.669 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.context.support.AbstractApplicationContext.refresh (AbstractApplicationContext.java:543)
01:16:58.676 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh (EmbeddedWebApplicationContext.java:122)
01:16:58.677 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.boot.SpringApplication.refresh (SpringApplication.java:693)
01:16:58.677 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.boot.SpringApplication.refreshContext (SpringApplication.java:360)
01:16:58.678 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.boot.SpringApplication.run (SpringApplication.java:303)
01:16:58.678 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.boot.SpringApplication.run (SpringApplication.java:1118)
01:16:58.680 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.boot.SpringApplication.run (SpringApplication.java:1107)
01:16:58.680 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at io.syndesis.example.Application.main (Application.java:13)
01:16:58.680 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
01:16:58.680 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
01:16:58.683 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
01:16:58.684 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at java.lang.reflect.Method.invoke (Method.java:498)
01:16:58.684 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at org.springframework.boot.maven.AbstractRunMojo$LaunchRunner.run (AbstractRunMojo.java:528)
01:16:58.684 [tc-okhttp-stream-788540094] INFO  INTEGRATION_RUNTIME_CONTAINER - STDOUT:     at java.lang.Thread.run (Thread.java:748)
```

--- a/app/common/util/src/main/java/io/syndesis/common/util/Json.java
+++ b/app/common/util/src/main/java/io/syndesis/common/util/Json.java
@@ -42,7 +42,8 @@ public final class Json {
     static {
         OBJECT_MAPPER = new ObjectMapper()
             .registerModules(new Jdk8Module())
-            .setDefaultPropertyInclusion(JsonInclude.Value.construct(JsonInclude.Include.NON_EMPTY, JsonInclude.Include.NON_EMPTY))
+            // keep using the deprecated method here in order to align with the jackson version (2.8.11) being used at integration runtime
+            .setPropertyInclusion(JsonInclude.Value.construct(JsonInclude.Include.NON_EMPTY, JsonInclude.Include.NON_EMPTY))
             .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
             .enable(DeserializationFeature.READ_ENUMS_USING_TO_STRING)
             .enable(SerializationFeature.WRITE_ENUMS_USING_TO_STRING)

